### PR TITLE
Bug 1963079: a smoke test for preferred host for KCM

### PR DESCRIPTION
**Pull request by p0lyn0mial**
The tests points the KCM to a non available host and sets `unsupported-kube-api-over-localhost` flag which changes it to use localhost instead. It then waits for new KCM and tests if creating an SA works.

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
requires https://github.com/openshift/library-go/pull/1011, https://github.com/openshift/cluster-kube-controller-manager-operator/pull/509 and https://github.com/openshift/kubernetes/pull/603

File: test/e2e-preferred-host/kcm_preferred_host_kas_test.go [hunk 1]
```diff
@@ -1,7 +1,101 @@
...
+	raw, err := json.Marshal(data)
+	require.NoError(t, err)
+	kcmOperatorConfig.Spec.UnsupportedConfigOverrides.Raw = raw
+	_, err = kcmOperator.Update(context.TODO(), kcmOperatorConfig, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	testlibrary.WaitForPodsToStabilizeOnTheSameRevision(t, kubeClient.CoreV1().Pods(operatorclient.TargetNamespace), "kube-controller-manager=true", waitForKCMRevisionSuccessThreshold, waitForKCMRevisionSuccessInterval, waitForKCMRevisionPollInterval, waitForKCMRevisionTimeout)
+
+	// act
+	t.Log("creating a namespace and waiting for a default sa to be populated")
+	testNs, err := k8se2eframework.CreateTestingNS("kcm-preferred-host", kubeClient, nil)
```
Comment (p0lyn0mial) [inline test/e2e-preferred-host/kcm_preferred_host_kas_test.go:74]: the `bump (*)` is big because we are pulling `k/k`

Comment (soltysh) [reply test/e2e-preferred-host/kcm_preferred_host_kas_test.go:74]: Can we not bring entire k/k to get just these 2 methods, but just copy them, they're not super hard to write nor sophisticated :cold_sweat:

**Review by soltysh — CHANGES_REQUESTED** [review]

File: test/e2e-preferred-host/kcm_preferred_host_kas_test.go (hunk 1 above)
Comment (p0lyn0mial) [reply test/e2e-preferred-host/kcm_preferred_host_kas_test.go]: done

**Comment by p0lyn0mial** [comment]
> ci/prow/e2e-aws-operator-preferred-host 

failed on `Error: unknown flag: --unsupported-kube-api-over-localhost`

mhm, I haven't changed the flag name, it is defined here https://github.com/openshift/kubernetes/pull/759/files#diff-d073b3d85bfc4877d354264d569dd3e19d544d1775a67f1650d661299869aa7aR276


maybe the test created a cluster with the old code. Trying one more time.

**Review by damemi — COMMENTED** [review]
/lgtm
/retest

**Pull request merged**